shapely/geometry/base.py

```diff
@@ -618,6 +618,14 @@ def within(self, other):
         """Returns True if geometry is within the other, else False"""
         return bool(shapely.within(self, other))
 
+    def dwithin(self, other, distance):
+        """
+        Returns True if geometry is within a given distance from the other, else False.
+
+        Refer to `shapely.dwithin` for full documentation.
```

Comment:
One thing I noticed in the generated docs (unless I'm building them incorrectly) is that the Shapely docs do not add a note about the required version of GEOS, whereas pygeos included a note about this.

Does this also need the `@requires_geos("3.10.0")` annotation here?

Reply:
no annotation is required, a helpful exception is raised: "shapely.errors.UnsupportedGEOSVersionError: 'dwithin' requires at least GEOS 3.10.0"

Reply:
Ah, that's a good point about the docstrings not including those notes. 
@mwtoews from pygeos, we had an automated injection of a note into the docstring about the required GEOS version, see eg https://pygeos.readthedocs.io/en/stable/constructive.html#pygeos.constructive.build_area vs https://shapely.readthedocs.io/en/latest/constructive.html#shapely.constructive.build_area 

The code for this is still included, so not really sure why this is not working .. (we also have tests for this, and those are passing)

Reply:
Fixing the docstring note in https://github.com/shapely/shapely/pull/1519